src/Processors/Formats/Impl/ORCBlockInputFormat.cpp

```diff
@@ -39,23 +38,53 @@ Chunk ORCBlockInputFormat::generate()
     if (!file_reader)
         prepareReader();
 
-    if (stripe_current >= stripe_total)
-        return res;
+    while (true)
+    {
+        if (!batch_reader)
+        {
+            arrow::Status reader_status = file_reader->NextStripeReader(
+                DBMS_DEFAULT_BUFFER_SIZE, include_indices, &batch_reader);
+            if (!reader_status.ok())
+                throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
+                                       "Failed to create batch reader: {}",
+                                       reader_status.ToString());
+        }
 
-    std::shared_ptr<arrow::RecordBatch> batch_result;
-    arrow::Status batch_status = file_reader->ReadStripe(stripe_current, include_indices, &batch_result);
-    if (!batch_status.ok())
-        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
-                               "Error while reading batch of ORC data: {}", batch_status.ToString());
+        if (!batch_reader)
+            break;
 
-    auto table_result = arrow::Table::FromRecordBatches({batch_result});
-    if (!table_result.ok())
-        throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
-                               "Error while reading batch of ORC data: {}", table_result.status().ToString());
+        std::shared_ptr<arrow::RecordBatch> batch_result;
+        arrow::Status batch_status = batch_reader->ReadNext(&batch_result);
+        if (!batch_status.ok())
+            throw ParsingException(ErrorCodes::CANNOT_READ_ALL_DATA,
+                                   "Error while reading batch of ORC data: {}",
+                                   batch_status.ToString());
+
+        if (!batch_result || !batch_result->num_rows())
```

Comment:
Why do we reset batch_reader in case of empty batch? Does ampty batch mean end of stream too?